src/java/org/apache/cassandra/index/sai/disk/v1/SSTableIndexWriter.java

```diff
@@ -327,7 +327,9 @@ private void writeSegmentsMetadata() throws IOException
 
     private SegmentBuilder newSegmentBuilder()
     {
-        SegmentBuilder builder = new SegmentBuilder.RAMStringSegmentBuilder(indexContext.getValidator(), limiter);
+        SegmentBuilder builder = TypeUtil.isLiteral(indexContext.getValidator())
```

Comment:
Not directly related to the changes, but I think that the method `TypeUtil.isUTF8OrAscii` invoked by `TypeUtil.isLiteral` could use `type instanceof StringType` instead of `type instanceof StringType || type instanceof AsciiType`.